repair/repair.cc

```diff
@@ -2344,19 +2348,30 @@ future<> repair::tablet_repair_task_impl::run() {
                 task_impl_ptr->neighbors = std::move(neighbors);
                 auto task = co_await rs._repair_module->make_task(std::move(task_impl_ptr), parent_data);
                 task->start();
-                co_await task->done();
+                auto res = co_await coroutine::as_future(task->done());
+                if (res.failed()) {
+                    auto ep = res.get_exception();
+                    // keyspace / table still exist?
+                    sstring dropped = co_await repair::table_sync_and_check(rs.get_db().local(), rs.get_migration_manager(), m.tid) ?
+                        (rs.get_db().local().has_keyspace(m.keyspace_name) ? "table" : "keyspace") : "";
```

Comment:
This is correct. 

However, I am not a fan of this hack of using empty string to express that the table is not dropped. 

Isn't the following much easier to read.
```
auto dropped = co_await repair::table_sync_and_check();
if (dropped) {
   bool ks_exists = rs.get_db().local().has_keyspace(m.keyspace_name);
   rtlogger.warn("...., The {} does not exist anymore, ....", ks_exist ? "table" : "keyspace");
} else {
    error =  std::move(ep)
}
```

Btw, we do not need above warning log for the case error = std::move(ep), to avoid log it twice,  the handler will log the error.

Reply:
Won't we lose information about which range failed if we only print the error at the end?
If there are multiple failed ranges I think we would like to know about all of them.
Although we can probably retrieve this information from other nodes, I think it's better if the repair master warns about all failed ranges as it's easier to track this way.

That said we can have both `bool dropped` and `sstring extra_error_msg` that will be set only in the `dropped` case similar to your suggestion above.

Reply:
When the first range fails, the loop ends. The second range will not be executed. There is nothing to lose.  invoke_on_all is supposed to collect all exceptions if more than one shard had exception.

Reply:
simplified in v4

Reply:
> When the first range fails, the loop ends. The second range will not be executed. There is nothing to lose. invoke_on_all is supposed to collect all exceptions if more than one shard had exception.

The whole point of this patch is to keep on going and attempt all ranges, even if any if them failed  so the loop continues, and we keep the error, to be returned only when the loop is done.